Files/UserControls/NavigationToolbar.xaml.cs

```diff
@@ -877,30 +877,34 @@ private void Refresh_Click(object sender, RoutedEventArgs e)
             RefreshRequested?.Invoke(this, EventArgs.Empty);
         }
 
-        private void SearchReigon_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs args)
+        private void SearchRegion_QuerySubmitted(AutoSuggestBox sender, AutoSuggestBoxQuerySubmittedEventArgs args)
         {
             SearchQuerySubmitted?.Invoke(sender, args);
         }
 
-        private void SearchReigon_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
+        private void SearchRegion_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
         {
             SearchTextChanged?.Invoke(sender, args);
         }
 
-        private void SearchReigon_SuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs args)
+        private void SearchRegion_SuggestionChosen(AutoSuggestBox sender, AutoSuggestBoxSuggestionChosenEventArgs args)
         {
             SearchSuggestionChosen?.Invoke(sender, args);
-            IsSearchReigonVisible = false;
+            IsSearchRegionVisible = false;
         }
 
         private void SearchButton_Click(object sender, RoutedEventArgs e)
         {
-            IsSearchReigonVisible = true;
+            IsSearchRegionVisible = true;
 
-            SearchReigon.Focus(FocusState.Programmatic);
+            // Given that binding and layouting might take a few cycles, when calling UpdateLayout
+            // we can guarantee that the focus call will be able to find an open ASB
+            SearchRegion.UpdateLayout();
```

Comment:
Theoretically, we could also wait for one or two rendering ticks, but that would require a bit more code. This is easier to read and probably more robust.

Reply:
I originally tried with a small delay, but it led to a bad experience, besides for needing less code, this is probably the better solution.